nodejs/eks/cluster.ts

```diff
@@ -223,14 +225,38 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("instanceRole and instanceRoles are mutually exclusive, and cannot both be set.");
     }
 
-    // If no VPC was specified, use the default VPC.
+    if (args.subnetIds && (args.publicSubnetIds || args.privateSubnetIds)) {
+        throw new Error("subnetIds, and the use of publicSubnetIds and/or privateSubnetIds are mutually exclusive. Choose a single approach.");
+    }
+
+    // Configure default networking architecture.
     let vpcId: pulumi.Input<string> = args.vpcId!;
     let subnetIds: pulumi.Input<pulumi.Input<string>[]> = args.subnetIds!;
```

Comment:
What is `subnetIds` used for now?

Reply:
The primary use of `subnetIds` is now only for:

1) backwards compatibility for users still on `subnetIds`:
2) and for use in the default VPC case when no `nodeSubnetIds` overrides are set, or if `publicSubnetIds` and/or `privateSubnetIds` is not set per the PR changes:

See:

https://github.com/pulumi/pulumi-eks/blob/18b55dd9dc6e03595d8eec3057f1eab1ae7892a5/nodejs/eks/cluster.ts#L240-L246

and

https://github.com/pulumi/pulumi-eks/blob/18b55dd9dc6e03595d8eec3057f1eab1ae7892a5/nodejs/eks/nodegroup.ts#L460-L474